ios/ContactPicker.m

```diff
@@ -43,24 +43,32 @@ - (instancetype)init {
     self.resolve = resolve;
     self.reject = reject;
     self.pickerController.peoplePickerDelegate = self;
-
+    
     if (ABAddressBookGetAuthorizationStatus() != kABAuthorizationStatusAuthorized) {
         ABAddressBookRequestAccessWithCompletion(self.addressBookRef, ^(bool granted, CFErrorRef error) {
             if (granted) {
-                [self.root presentViewController:self.pickerController animated:YES completion:nil];
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    UIViewController *root = [[[[UIApplication sharedApplication] delegate] window] rootViewController];
+                    [root presentViewController:self.pickerController animated:YES completion:nil];
+                });
             } else {
                 self.reject(@"access_denied", @"We need access to your contacts to use this feature, please go into your settings and enable it.",(__bridge NSError *)error);
             }
         });
     } else {
-      [self.root presentViewController:self.pickerController animated:YES completion:nil];
+        dispatch_async(dispatch_get_main_queue(), ^{
+            UIViewController *root = [[[[UIApplication sharedApplication] delegate] window] rootViewController];
+            [root presentViewController:self.pickerController animated:YES completion:nil];
+        });
     }
 }
 - (void)peoplePickerNavigationController:(ABPeoplePickerNavigationController *)peoplePicker
                          didSelectPerson:(ABRecordRef)person {
     ABMutableMultiValueRef emailProperty  = ABRecordCopyValue(person, kABPersonEmailProperty);
     NSArray *emails = (__bridge_transfer NSArray *)ABMultiValueCopyArrayOfAllValues(emailProperty);
-    [self.root dismissViewControllerAnimated:YES completion:nil];
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self.root dismissViewControllerAnimated:YES completion:nil];
```

Comment:
I think you're going to need `UIViewController *root = [[[[UIApplication sharedApplication] delegate] window] rootViewController];` here too, and just get rid of `self.root`, so you make sure you have the right root controller when you need it.